pennylane/gradients/parameter_shift.py

```diff
@@ -628,32 +631,42 @@ def processing_fn(results):
 
         multi_measure = len(tape.measurements) > 1
         if multi_measure:
-
-            res = []
-            for idx, m in enumerate(mask):
-
-                if isinstance(pdA2, (Sequence, np.ndarray)):
-                    r = _get_var_with_second_order(pdA2[idx], f0[idx], pdA[idx]) if m else pdA[idx]
-                elif (
-                    isinstance(pdA, tuple)
-                    and len(pdA) > 0
-                    and isinstance(pdA, (Sequence, np.ndarray))
-                ):
-                    r = (
-                        [
-                            _get_var_with_second_order(pdA2, f0[idx], pdA[idx][i])
-                            for i in range(len(pdA))
-                        ]
-                        if m
-                        else [pdA[idx][i] for i in range(len(pdA))]
-                    )
-
-                res.append(r)
-            return tuple(res)
+            num_params = len(tape.trainable_params)
+
+            if num_params > 1:
+                var_grad = []
+                for m_idx in range(len(tape.measurements)):
+                    m_res = []
+                    measurement_is_var = mask[m_idx]
+                    if measurement_is_var:
+                        for p_idx in range(num_params):
+                            _pdA2 = pdA2[m_idx][p_idx] if pdA2 != 0 else pdA2

```

Comment:
Minor: can we change the "dummy" return value of `_get_pdA2` to be `None` instead of `0`? This line reads a bit confusing in the latter case.

Reply:
Unless necessary, I wouldn't do that because the second derivative in such a case is 0. Any reason why `None` would work better?